rust/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -480,48 +484,84 @@ struct OperationBatch {
 
 impl OperationBatch {
     async fn submit(self, confirm_queue: &mut OpQueue, metrics: &SerialSubmitterMetrics) {
-        match self.try_submit_as_batch(metrics).await {
-            Ok(outcome) => {
-                info!(outcome=?outcome, batch_size=self.operations.len(), batch=?self.operations, "Submitted transaction batch");
-                let total_estimated_cost = total_estimated_cost(&self.operations);
-                for mut op in self.operations {
-                    op.set_operation_outcome(outcome.clone(), total_estimated_cost);
-                    op.set_next_attempt_after(CONFIRM_DELAY);
-                    confirm_queue
-                        .push(op, Some(PendingOperationStatus::Confirm(SubmittedBySelf)))
-                        .await;
-                }
-                return;
+        let excluded_ops = match self.try_submit_as_batch(metrics).await {
+            Ok(batch_result) => {
+                Self::handle_batch_result(self.operations, batch_result, confirm_queue).await
             }
             Err(e) => {
                 warn!(error=?e, batch=?self.operations, "Error when submitting batch. Falling back to serial submission.");
+                self.operations
             }
+        };
+
+        if !excluded_ops.is_empty() {
+            debug!(excluded_ops=?excluded_ops, "Operations would revert in batch. Submitting individually.");
```

Comment:
nit: seems like the Err case above will result in two logs:
```
Error when submitting batch. Falling back to serial submission.
Operations would revert in batch. Submitting individually.
```
Do we want the second log if the first one's printed? could be a little confusing

Reply:
sgtm!